src/drivers/driver.ts

```diff
@@ -69,6 +70,12 @@ export abstract class CMakeDriver implements vscode.Disposable {
    * @returns The exit code from CMake
    */
   protected abstract doConfigure(extra_args: string[], consumer?: proc.OutputConsumer): Promise<number>;
+  protected abstract doCacheConfigure(): Promise<number>;
+
+  protected _isConfiguredAtLeastOnce = false;
+  protected isConfiguredAtLeastOnce(): boolean {
```

Comment:
nit: you can make this a 'get' property so that you don't have to access it like a function (i.e. no parenthesis)